### PR TITLE
Upgrade to Cloud Hypervisor v27.0

### PR DESCRIPTION
**Pull request by likebreath**
This release has been tracked in our new [roadmap project ](https://github.com/orgs/cloud-hypervisor/projects/6) as iteration v27.0.

**Community Engagement**
A new mailing list has been created to support broader community discussions.
Please consider [subscribing](https://lists.cloudhypervisor.org/g/dev/); an announcement of a regular meeting will be 
announced via this list shortly.

**Prebuilt Packages**
Prebuilt packages are now available. Please see this [document](https://github.com/cloud-hypervisor/obs-packaging/blob/main/README.md)
on how to install. These packages also include packages for the different
firmware options available.

**Network Device MTU Exposed to Guest**
The MTU for the TAP device associated with a virtio-net device is now exposed
to the guest. If the user provides a MTU with --net mtu=.. then that MTU is
applied to created TAP interfaces. This functionality is also exposed for
vhost-user-net devices including those created with the reference backend.

**Boot Tracing**
Support for generating a trace report for the boot time has been added
including a script for generating an SVG from that trace.

**Simplified Build Feature Flags**
The set of feature flags, for e.g. experimental features, have been simplified:

* msvh and kvm features provide support for those specific hypervisors
(with kvm enabled by default),
* tdx provides support for Intel TDX; and although there is no MSHV support
now it is now possible to compile with the mshv feature,
* tracing adds support for boot tracing,
* guest_debug now covers both support for gdbing a guest (formerly gdb
feature) and dumping guest memory.

The following feature flags were removed as the functionality was enabled by
default: amx, fwdebug, cmos and common.

**Asynchronous Kernel Loading**
AArch64 has gained support for loading the guest kernel asynchronously like
x86-64.

**GDB Support for AArch64**
GDB stub support (accessed through --gdb under guest_debug feature) is now
available on AArch64 as well as as x86-64.

**Notable Bug Fixes**
* This version incorporates a version of virtio-queue that addresses an issue
where a rogue guest can potentially DoS the VMM,
* Improvements around PTY handling for virtio-console and serial devices,
* Improved error handling in virtio devices.

**Deprecations**
Deprecated features will be removed in a subsequent release and users should
plan to use alternatives.

* Booting legacy firmware (compiled without a PVH header) has been deprecated.
All the firmware options (Cloud Hypervisor OVMF and Rust Hypervisor Firmware)
support booting with PVH so support for loading firmware in a legacy mode is no
longer needed. This functionality will be removed in the next release.

Details can be found: https://github.com/cloud-hypervisor/cloud-hypervisor/releases/tag/v27.0

Fixes: https://github.com/kata-containers/kata-containers/issues/5309

Signed-off-by: Bo Chen <chen.bo@intel.com>

### CODE REVIEW

**Comment by likebreath** [comment]
This is a draft PR pending on https://github.com/cloud-hypervisor/cloud-hypervisor/pull/4720.

/test

**Comment by likebreath** [comment]
@fidencio Please take a look. Here is the the updated way for booting with td-shim: https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/docs/intel_tdx.md#tdshim.

**Comment by likebreath** [comment]
Note: The CI job `jenkins-ci-ARM-ubuntu-18-04` has been consistently failing for the past several days, and its failure is not related to the changes here.

**Comment by fidencio** [comment]
> @fidencio Please take a look. Here is the the updated way for booting with td-shim: https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/docs/intel_tdx.md#tdshim.

The changes related to TDX do make sense to me, thanks @likebreath!

I see this PR is pointing to a specific commit, should we update it to the released version?

**Comment by likebreath** [comment]
> I see this PR is pointing to a specific commit, should we update it to the released version?

@fidencio A specific commit revision (after the v27.0 release) is needed as we need to include the fix from https://github.com/cloud-hypervisor/cloud-hypervisor/pull/4720. 

I also put a note in the first commit message. Please let me know if it needs further clarification. 

```
Note: To have the new API of loading firmware for booting (e.g. boot
from td-shim), a specific commit revision after the v27.0 release is
used as the Cloud Hypervisor version from the 'versions.yaml'.
```

**Comment by fidencio** [comment]
> Member

No, that's totally fine and I missed that.
Thanks, @likebreath!

File: versions.yaml [hunk 1]
```diff
@@ -75,7 +75,7 @@ assets:
       url: "https://github.com/cloud-hypervisor/cloud-hypervisor"
       uscan-url: >-
         https://github.com/cloud-hypervisor/cloud-hypervisor/tags.*/v?(\d\S+)\.tar\.gz
-      version: "v26.0"
+      version: "2115a4156891804e5fc7bbd0d1831d2e92a0c50e"
```
Comment (amshinde) [inline versions.yaml:78]: why aren't you using the v27.0 tag here?

**Review by amshinde — APPROVED** [review]

**Pull request merged**